api/lib/opentelemetry/trace/samplers/always_sample_sampler.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/trace/samplers/basic_decision'
+
+module OpenTelemetry
+  module Trace
+    module Samplers
+      # The {AlwaysSampleSampler} always returns a true sampling decision.
+      class AlwaysSampleSampler
+        ALWAYS_SAMPLE_DECISION = BasicDecision.new(decision: true)
+
+        # Returns the sampling {Decision} for a {Span} to be created
+        #
+        # @param [SpanContext] span_context The {SpanContext} of a parent span,
+        #   typically extracted from the wire. Can be nil.
+        # @param [Boolean] extracted_context True if span_context was extracted
+        #   from the wire. Can be nil.
+        # @param [Integer] trace_id The trace_id of the {Span} to be created
+        # @param [Integer] span_id The span_id of the {Span} to be created
+        # @param [String] span_name Name of the {Span} to be created
+        # @param [Array<Links>] links A collection of links to be associated
```

Comment:
Do we want to be this specific with the type? I.e. `Array<...>`

Reply:
It feels like `nil` should be allowed here. We often won't have any Links, and we shouldn't require the allocation of an empty array just to call this method.

Reply:
This is also unclear from the spec. Cross referencing the Java Implementation, `parentLinks` are not nullable on the AlwaysSampleSampler or NeverSampleSampler. 
AlwaysSampleSampler:
https://github.com/open-telemetry/opentelemetry-java/blob/6d1b31bfa639af653649d694ea5660fb7c3fd5ce/api/src/main/java/io/opentelemetry/trace/util/Samplers.java#L71-L81

NeverSampleSampler
https://github.com/open-telemetry/opentelemetry-java/blob/6d1b31bfa639af653649d694ea5660fb7c3fd5ce/api/src/main/java/io/opentelemetry/trace/util/Samplers.java#L100-L108

But they are on the ProbabilitySampler:

https://github.com/open-telemetry/opentelemetry-java/blob/6d1b31bfa639af653649d694ea5660fb7c3fd5ce/sdk/src/main/java/io/opentelemetry/sdk/trace/samplers/ProbabilitySampler.java#L76-L83

Reply:
I changed this to an `Enumerable<Link>`, that can be nil. The nil part is subject to the clarification on the spec. Does the `Enumerable<Link>` work for you?